joomla/joomla-cms
# [4.0] [CSP] Initial Integration of com_csp & plg_system_httpheaders

### Summary of Changes

This PR is the next step for the implementation of a core csp handling.
This now does the integration of com_csp & plg_system_httpheaders.

The general goal of the PR is that everything that is published in com_csp also gets added to the automatic csp rule including some general things like nonce's

### Testing Instructions

##### Testing the `detecting` mode
- Go to the Plugin
- Got to the CSP tab in the plugin settings
- Set the `mode` to `detecting`
- Visit the frontend & navigate in the backend
- check the com_csp backend
- notice the collected reports also with client info

##### Testing the `custom` mode

- Go to the Plugin
- Got to the CSP tab in the plugin settings
- Set the `mode` to `custom`
- Notice that now the additional fields open
- configure (i would suggest) a report only rule
- confirm using the browser's networktab that the header is set

##### Testing the `auto` mode #1

- make sure NO report is published in com_csp
- Go to the Plugin
- Got to the CSP tab in the plugin settings
- Set the `mode` to `auto`
- confirm using the browser's networktab that NO CSP header is set

##### Testing the `auto` mode #2

- Go to com_csp
- publish all collected rules
- Go to the Plugin
- Got to the CSP tab in the plugin settings
- Set the `mode` to `auto`
- confirm using the browser's networktab that the CSP header IS set

### Expected result

All four test cases works

### Actual result

No integration between com_csp & plg_system_httpheaders

### Documentation Changes Required

This is a new feature so docu is needed
